src/app/budget/[month]/components/PageHeader.tsx

```diff
@@ -5,11 +5,15 @@ interface PageHeaderProps {
 }
 
 export function PageHeader({ month }: PageHeaderProps) {
+  // Parse the month string properly
+  const [year, monthNum] = month.split("-");
+  const date = new Date(parseInt(year), parseInt(monthNum) - 1);
+
   return (
     <div className="flex justify-between items-center mb-6">
       <h1 className="text-2xl font-bold">
         Budget for{" "}
-        {new Date(month + "-01").toLocaleString("default", {
+        {date.toLocaleDateString("default", {
```

Comment:
it's better to create utility function and make it reusable. Without it code become unconsistent very soon